### PR TITLE
VZ Map: add 2021 SSZ Google Sheet to airflow process

### PR DESCRIPTION
**Pull request by KatiRG**
## What this pull request accomplishes:

- updates script `bdit_data-sources/vision_zero/schools.py` to read from 2021 Google sheet
- updates dag `bdit_data-sources/dags/vz_google_sheets.py` with `task4` that calls `pull_from_sheet` for 2021

## Issue(s) this solves:
#392 

## What, in particular, needs to reviewed:

- You can have a look at the added code but the real check would be to run the dag

### CODE REVIEW

**Review by radumas — CHANGES_REQUESTED** [review]
No issues with the code itself. I take it you actually tested pulling the 2021 sheet in isolation?

**Comment by radumas** [comment]
Oh I guess CodeFactor doesn't like that the file doesn't end with an empty newline.

**Comment by KatiRG** [comment]
> 
> 
> No issues with the code itself. I take it you actually tested pulling the 2021 sheet in isolation?

Yes 👍

File: vision_zero/schools.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """ This script reads 3 Vision Zero google spreadsheets ('2018 School Safety Zone', '2019 School Safety Zone',
```
[the remark this answers is not in the document]
Comment (KatiRG) [reply vision_zero/schools.py:1]: done

File: vision_zero/README.md [hunk 2]
```diff
@@ -58,6 +58,6 @@ from the Google Sheets and put them into postgres tables with the following fiel
 * The script being used reads up to line 180 although the actual data is less than that. This is to anticipate extra schools which might be added into the sheets in the future.
```
[the remark this answers is not in the document]
Comment (KatiRG) [reply vision_zero/README.md:59]: done (according to the current method)

File: vision_zero/create-table-school_safety_zone_2021_raw.sql [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* Create empty table to store data from 2021 SSZ Google Sheet
+*/
+CREATE TABLE vz_safety_programs_staging.school_safety_zone_2021_raw
```
[the remark this answers is not in the document]
Comment (KatiRG) [reply vision_zero/create-table-school_safety_zone_2021_raw.sql:3]: Do you mean the view that selects from all these tables would not need separate CTEs? This is the skeleton of what we have now:
```
WITH staging2018 AS (
	SELECT *
	FROM vz_safety_programs_staging.school_safety_zone_2018_raw a
	WHERE locations_zone IS NOT NULL
)
, staging2019 AS (
	SELECT *
	FROM vz_safety_programs_staging.school_safety_zone_2019_raw b
	WHERE locations_zone IS NOT NULL
)
, staging2020 AS (
	SELECT *
	FROM vz_safety_programs_staging.school_safety_zone_2020_raw c
	WHERE locations_zone IS NOT NULL
)
, staging2021 AS (
	SELECT *
	WHERE locations_zone IS NOT NULL
)
```

**Review by radumas — CHANGES_REQUESTED** [review]
Are you going to consolidate these three dags? They don't all need to be merged to main.

Also the config **should not** be committed to GitHub

**Comment by KatiRG** [comment]
> 
> 
> Are you going to consolidate these three dags? They don't all need to be merged to main.
There should only be one dag  now, `vz_google_sheets.py`. I removed the test dag `vz_google_sheets_var.py`.
> 
> Also the config **should not** be committed to GitHub
ah yes! it's deleted now

File: vision_zero/create-table-school_safety_zone_2021_raw.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (KatiRG) [reply vision_zero/create-table-school_safety_zone_2021_raw.sql]: ok got it! I made a parent table now and defined each year table as a child

**Review by radumas — CHANGES_REQUESTED** [review]
One comment on the sql. Otherwise looks good. Great instructions.

File: vision_zero/create-table-school_safety_zone_2021_raw.sql [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/* Create parent table for all school safety zone sheets
+*/
+CREATE TABLE vz_safety_programs_staging.school_safety_zone_raw_parent (
```
[the remark this answers is not in the document]
Comment (KatiRG) [reply vision_zero/create-table-school_safety_zone_2021_raw.sql]: Makes sense. The file now has just the query for creating the 2021 staging table. I moved the query to define the parent table into the vz programs repo.

**Review by radumas — CHANGES_REQUESTED** [review]
I didn't realize there wasn't any sql stored in this folder. which begs the question, why include only the 2021 SSZ creation query in this PR, seems extraneous if it's defined elsewhere

**Comment by KatiRG** [comment]
> 
> 
> I didn't realize there wasn't any sql stored in this folder. which begs the question, why include only the 2021 SSZ creation query in this PR, seems extraneous if it's defined elsewhere

True that the parent query is elsewhere. But if you need to add a new spreadsheet you need to follow the steps in this repo so I thought it was better to avoid having to hop over to another repo just to get started (making the empty table is really the first thing you have to do).

The query to make the new table is written in the [README](https://github.com/CityofToronto/bdit_data-sources/tree/ssz2021/vision_zero#31-create-empty-table-in-database-to-store-spreadsheet) of this repo so if you think this is sufficient then I can delete the actual sql file `create-table-school_safety_zone_2021_raw.sql`.

**Comment by radumas** [comment]
Yeah I think the steps in the README are clear enough so the file is unnecessary. Thanks.

**Review by radumas — APPROVED** [review]

**Pull request merged**